keyboards/dichotomy/keymaps/default/keymap.c

```diff
@@ -0,0 +1,518 @@
+// this is the style you want to emulate.
+// This is the canonical layout file for the Quantum project. If you want to add another keyboard,
+
+#include "dichotomy.h"
+#include "report.h"
+#include "pointing_device.h"
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+enum dichotomy_layers
+{
+	_BS,
+	_SF,
+	_NM,
+	_NS,
+	_MS
+};
+
+#define LONGPRESS_COUNT 4
+
+enum dichotomy_keycodes 
+{
+  CK_1G = SAFE_RANGE,
+  CK_BSPE,
+  CK_QE,
+  CK_TE, //these 4 CK_XXXX keys are special "alternate long-press" keys controlled with unique timers.  Make sure you understand them before you mess with them.
+  NS_HYPH,
+  NS_EQU,
+  NUMKEY,
+  SFTKEY,
+  MOUKEY,
+  MS_BTN1,
+  MS_BTN2,
+  MS_BTN3
+};
+
+// Macro definitions for readability
+enum dichotomy_macros
+{
+	VOLU,
+	VOLD,
+	ESCM
+};
+
+#define LONGPRESS_DELAY 150
+#define MAX_TOGGLE_LENGTH 300
+#define TAPPING_TOGGLE 1
```

Comment:
This needs to be in the `config.h` file, actually.

Reply:
These are actually specific to the keymap, not using a traditional tap dance.

Reply:
Well, if you wanted to keep it in the keymap.c, then I would recommend a different name, then.  That way there isn't a conflict in the case that somebody uses the value in userspace, or the like.